feat: add `dbg!()` error reports when processing validation

jrconlin opened the pull request:
Hawk currently just bails on error meaning that determining what went
wrong can be difficult. I've found adding a few `dbg!()` statements can
help quickly isolate problems in test cases, setup, initial connection
tests, etc.

jrconlin committed:
feat: add `dbg!()` error reports when processing validation

Hawk currently just bails on error meaning that determining what went
wrong can be difficult. I've found adding a few `dbg!()` statements can
help quickly isolate problems in test cases, setup, initial connection
tests, etc.

djmitche reviewed (commented):
TIL `dbg!`. 

> Note that the macro is intended as a debugging tool and therefore you should avoid having uses of it in version control for longer periods. Use cases involving debug output that should be added to version control are better served by macros such as debug! from the log crate.

That suggests to me that this shouldn't be here permanently, since it will be spewing these messages to stderr in a production service.  Am I missing something?

jrconlin commented:
`dbg!()` shouldn't dump messages, unless the server is configured to do so with something like `RUST_LOG=debug cargo run` Normally they just don't appear. They can show up in some unit tests, so that's a potentially valid objection. 

I'll redo this to use the `debug!()` macro. Thanks!

jrconlin committed:
f convert `dbg!` to `debug!`

NOTE: this will require an externally defined logger such as
[`env_logger`](https://docs.rs/env_logger/0.6.2/env_logger/) initialized
in the calling program.

jrconlin committed:
f fmt

djmitche reviewed (approved):
`log::debug!()` seems like a fine choice :)

> dbg!() shouldn't dump messages, unless the server is configured to do so with something like RUST_LOG=debug cargo run Normally they just don't appear. They can show up in some unit tests, so that's a potentially valid objection.

I'm missing something here :)

```rust
fn main() {
    dbg!("Hello, world!");                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     
}
```

does indeed output `[src/main.rs:2] "Hello, world!" = "Hello, world!"` without any special setting of `RUST_LOG`.  The docs for `dbg!` don't say anything about `RUST_LOG`, either.

jrconlin commented:
Huh, weird! for me `dbg!()` never prints out unless I either run in the debugger or have the `RUST_LOG` setting. I wonder why mine is different? Anyway, great call on asking me to switch to normal logging. No need to bother folks that don't need to be bothered.

Thanks for the merge!
